src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1479 @@
+// Copyright 2015 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::cell::RefCell;
+use std::cmp::PartialEq;
+use std::default::Default;
+use std::ffi::CString;
+use std::io::{Error, ErrorKind};
+use std::marker::{Send, Sync};
+use std::mem;
+use std::ops::Deref;
+use std::env;
+use std::ptr;
+use std::slice;
+
+use libc::{intptr_t, memcmp};
+
+use uuid::Uuid;
+
+use serde;
+use bincode;
+use bincode::serde::DeserializeError;
+
+use winapi;
+use winapi::{HANDLE, INVALID_HANDLE_VALUE, LPVOID};
+use kernel32;
+
+lazy_static! {
+    static ref CURRENT_PROCESS_ID: winapi::ULONG = unsafe { kernel32::GetCurrentProcessId() };
+    static ref CURRENT_PROCESS_HANDLE: intptr_t = unsafe { kernel32::GetCurrentProcess() as intptr_t };
+
+    static ref DD_ENABLED: bool = match env::var_os("DD") {
+        Some(_) => true,
+        None => false,
+    };
+    static ref DD2_ENABLED: bool = match env::var_os("DD2") {
+        Some(_) => true,
+        None => false,
+    };
+}
+
+macro_rules! dd { ($($rest:tt)*) => { if *DD_ENABLED { println!($($rest)*); } } }
+macro_rules! dd2 { ($($rest:tt)*) => { if *DD_ENABLED || *DD2_ENABLED { println!($($rest)*); } } }
+
+// We stick these in WinErrors to indicate various things
+// that aren't quite errors.
+const MAGIC_CODE_BASE: u32 = 0xdeadbee0u32;
+const MAGIC_NO_OUTSTANDING_DATA_CODE: u32 = MAGIC_CODE_BASE + 0;
+const MAGIC_CHANNEL_CLOSED_CODE: u32 = MAGIC_CODE_BASE + 1;
```

Comment:
Hmm, I could try? I guess I can make WinError be an enum of one of these, or an actual Windows error code with a code.  Didn't think of that, will do.